src/app/dashboard/dashboard-story.component.ts

```diff
@@ -0,0 +1,108 @@
+import { Component, Input, OnInit } from '@angular/core';
+import { StoryModel, SeriesModel } from '../shared';
+import { Env } from '../core/core.env';
+
+@Component({
+  selector: 'publish-dashboard-story',
+  styleUrls: ['dashboard-story.component.css'],
+  template: `
+    <div *ngIf="status" class="{{status}} status bar"></div>
+    <p class="story-date">{{storyDate | date:"M/d"}}</p>
+    <div class="title">
+      <h2>
+        <a *ngIf="storyTitle" [routerLink]="editLink">{{storyTitle}}</a>
+        <a *ngIf="!storyTitle" [routerLink]="editLink">{{storyDate | date:"MMM d, y"}}</a>
+      </h2>
+      <h3>
+        <span class="teaser">{{storyTeaser}}</span>
+        <span *ngIf="status" class="{{status}} status text">{{status}}</span>
+      </h3>
+    </div>
+    <div class="actions">
+      <prx-spinner *ngIf="episodeLoader || podcastLoader" inverse="true"></prx-spinner>
+      <a *ngIf="metricsUrl" [href]="metricsUrl" rel="noopener noreferrer" target="_blank">
+        <span class="icon-bar-chart"></span>
+      </a>
+      <a [routerLink]="editLink"><span class="icon-pencil"></span></a>
+    </div>
+  `
+})
+
+export class DashboardStoryComponent implements OnInit {
+
+  @Input() series: SeriesModel;
+  @Input() story: StoryModel;
+  @Input() episodeLoader: boolean;
+  @Input() podcastLoader: boolean;
+
+  editLink: any[];
+
+  storyTitle: string;
+  storyTeaser: string;
+  storyDate: Date;
+
+  status: string;
+  metricsUrlParams: string;
+
+  ngOnInit() {
+    this.setLink();
+    this.setStatus();
+    this.loadData();
+    this.setMetricsUrlParams();
+  }
+
+  setLink() {
+    if (this.story.isNew) {
+      this.editLink = ['/story/new'];
+      if (this.story.parent) {
+        this.editLink.push(this.story.parent.id);
+      }
+    } else {
+      this.editLink = ['/story', this.story.id];
+    }
+  }
+
+  setStatus() {
+    if (this.story.isNew) {
+      this.status = 'new';
+    } else if (!this.story.publishedAt) {
+      this.status = 'draft';
+    } else if (!this.story.isPublished()) {
+      this.status = 'scheduled';
+    } else {
+      this.status = 'published';
+    }
+  }
+
+  loadData() {
```

Comment:
This method name gives the impression that this data is being fetched from a remote source.

Reply:
yeah... honestly there are some weird conventions and old bad habits in this code that I'm refactoring a little bit here. Stay tuned for an update! All this came from the existing home component stuff since that's where I initially started building this (instead of building new components from the start.)